### PR TITLE
exit early from the entrypoint script on error

### PR DESCRIPTION
**Pull request by girishmg**
The process_ready() function read's the process pid from the pidfile and
ensures that the pid is running. If the pidfile is not present, then it
keeps checking (through wait_for_event code) for the pidfile to show up.

However, there is a possibility that the process crashed and no pidfile
was written. Now, the above logic will keep trying for 400s and give up.
The status of the container will be Running but the process has died
within the first few seconds.

### CODE REVIEW

File: dist/images/ovnkube.sh [hunk 1]
```diff
@@ -225,6 +225,28 @@ ovs_ready () {
   return 0
 }
 
+# Very similar to process_ready function below with an additional check for
+# local processes. It also checks if the process is actually running and has
```
[the remark this answers is not in the document]
Comment (danwinship) [reply dist/images/ovnkube.sh]: Ah... I was interpreting "local" as meaning "on the same machine", not "in the same container". Maybe clarify in the comments on the functions.

**Comment by girishmg** [comment]
@danwinship @dcbw can you PTAL? I am using a different approach to exit early.

**Comment by danwinship** [comment]
makes sense. maybe have a `wait_for_local_process_ready` wrapper?

**Comment by girishmg** [comment]
@danwinship what if we need to change the number of retries on a per-process basis? The current code will accommodate that use case.

**Comment by danwinship** [comment]
ok then. lgtm

**Pull request merged**